### ISSUE TITLE
Previous session lost (open tabs) - how to restore?

### ISSUE THREAD

**Issue by silent5am** [brave/brave-browser#4416]
## Description 
Last session (open tabs) was lost after a regular reboot; unable to restore that session

## Steps to Reproduce
Can not reproduce at this point

## Brave version (brave://version info)
Version 0.63.55 Chromium: 74.0.3729.131 (Official Build) (64-bit)

## Version/Channel Information:
current release

I'm sure this is a known issue but I wasn't able to figure out how to fix this myself. 

I'm aware that there are session files stored both in the main directory, as well as the session storage folder (AppData\Local\BraveSoftware\Brave-Browser\User Data), however I'm not sure what to do with those files.

Firefox allows to rename those backup files and replace them with the flawed session, but this doesn't seem to work in this case (respectively not sure what I'm doing wrong).

I have old session files (.ldb) from last night that should contain my open tabs (afaik) but what do I do with them?

The sync function doesn't help because the other device was also updated to a brandnew session :(

Using the browser history is one way to "fix" this (quite tedious task tbh); copy/pasting a single file might be a bit more easier.

**Comment by silent5am** [comment]
Despite not being able to restore my previous session, I decided to make a copy of the entire folder.

Today, I had the exact same issue: I started Brave, the browser insta crashes. When I opened it again, all my tabs were gone.

So I copied the folder from last week (the "backup") into the main directory and replaced all new files with the old ones. However, something seems to auto-update or auto-overwrite the old files and replace them with the corrupted "last session" or "last tabs" files.

How do I make a proper backup? What folders/files are relevant to restore the last browser session (all open tabs, etc)? 

I've been comparing snapshots of the directory and trying to figure out what is what - two hours wasted so far.

It would be really nice to get some insight here.

**Comment by rebron** [comment]
@silent5am  

A few questions: 
Are you losing your cookies as well? 
When you run into this issue, can you restore your tabs under history where it says recently closed? Did the restore tabs popup show up also in the upper right hand corner when you opened the browser again?

**Comment by poikilos** [comment]
Did you remove any fonts at any point since the last time it loaded correctly before each crash (_and do you have the setting for On Startup: Continue where you left off_)?

**Comment by mixedCase** [comment]
Hi, I've had this happen to me twice now. I do have "Continue where you left off" set and no, cookies are not lost, neither is history. In fact, the browser remembers I had two windows with different profiles open; it's just that the tabs for each window are *gone*.

This is honestly extremely awful for my workflow and if it happens again I'll just have to switch to another Chromium fork since it's too unreliable as a daily driver for work.

**Comment by Nerothos** [comment]
This happen to me often on two computers too (Win10 both). Nothing in the window history remain so I can't restore a session, only the browser history.

**Comment by kjozwiak** [comment]
@mixedCase & @Nerothos when this usually happens, are you guys a few versions behind when upgrading or always on the current one? We've had a few reports in the past where users who miss a few releases and update sometimes run into this issue as well. Just curious if this could be the same situation or something else.

**Comment by mixedCase** [comment]
@kjozwiak I'm going to test that later, but last time it happened it wasn't the case. I maintain one of the Brave packaging scripts for Arch Linux and tend to update to stable releases the same day they are marked as such.

As a point probably worth noting, I believe last time it happened it was either first or second execution of the browser after updating, so if there's logic running to migrate the configuration  schema and there was a schema change with last version it is worth investigating there.

Regardless, updating through arbitrary versions without losing state should be possible within reason. I'm sure the Chromium team has a very generous period before they break things.

**Comment by Nerothos** [comment]
My versions are Version 0.65.121 and Version 0.66.100

**Comment by poikilos** [comment]
My previous comment may be invalid. I got the crash again without changing any fonts, but after removing nautilus, if that matters. I am still using The Great Suspender, but have been for many months and only ever got the instant disappearing silent crash and loss of tabs twice so far.

**Comment by poikilos** [comment]
I was referring to Gitea Version: 1.8.1, Go1.12.5

**Comment by kjozwiak** [comment]
@poikilos mind going into `brave://crashes`, submitting the crash reports and reporting the `Crash Report ID`?  Example of a submitted crash report:

<img width="688" alt="Screen Shot 2019-07-21 at 8 23 31 PM" src="https://user-images.githubusercontent.com/2602313/61599275-6fa60800-abf5-11e9-9223-43d348ca5653.png">

**Comment by Nerothos** [comment]
I got this when I try to report then.

![image](https://user-images.githubusercontent.com/3739096/61628682-48961780-ac83-11e9-90e9-ff9ab4e6527b.png)

Any thing that I have to do?

**Comment by kjozwiak** [comment]
You should be seeing a `Send now` link at the bottom of each crash, example:

<img width="508" alt="Screen Shot 2019-07-22 at 9 30 51 PM" src="https://user-images.githubusercontent.com/2602313/61675905-0ba04400-acc8-11e9-9681-0b68c48a8073.png">

Maybe you can enable `Automatically send crash reports to Brave` via `chrome://settings/privacy`?

**Comment by Nerothos** [comment]
Apparently it uploaded them afterwards.

**Comment by kjozwiak** [comment]
@Nerothos great 👍 Mind sharing the `Crash Report ID` for those crashes?

**Comment by Nerothos** [comment]
I'm not sure they are all relevant but here :
[crashid.txt](https://github.com/brave/brave-browser/files/3422633/crashid.txt)

**Comment by poikilos** [comment]
@kjozwiak Like before, crash reporting is disabled, and there are 0 crashes. Again, I wish I could enable reports but disable sending them automatically. I enabled them for now so I'll let you know if I get anything.

**Comment by kjozwiak** [comment]
> I'm not sure they are all relevant but here :
> [crashid.txt](https://github.com/brave/brave-browser/files/3422633/crashid.txt)

Thanks 👍I'll ask a dev to take a look at them.

**Comment by Annihilannic** [comment]
I have lost all my tabs in a similar way on at least 2 occasions.  I'm running under Windows 10.  While the issue is ongoing, is there a way we can back up our session?  It was apparently possible under previous versions (see https://github.com/brave/browser-laptop/issues/13483), but when I try to do so with version 0.69.135 some presumably critical files like 'Current Session' and 'Current Tabs' under 'C:\Users\userid\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default' are inaccessble while the browser is open.

**Comment by Voronovskiy** [comment]
If you are using Chrome you can restore your tabs following these steps [https://rocketfiles.com/articles/how-to-restore-tabs-on-chrome](https://rocketfiles.com/articles/how-to-restore-tabs-on-chrome).

**Comment by poikilos** [comment]
@Voronovskiy, The problem is that restoring does not work after some type(s) of crash(es). The second "solution" at the link proposes looking in history for all of your tabs, which is not a solution. Some of my tabs could be for a project I started a month ago or more, and regardless I open tabs over the course of several days. The only people who care about this issue are people who use tabs heavily. We need to enable crash reports, which is not on by default, then somehow hope to lose our data so we at least have a matching crash report number.

**Comment by NumDeP** [comment]
Any one doing important work and spend quite a bit of effort right-clicking with dozens of tabs open on the bar. I recommend this extension called SessionBuddy, it helps in these circumstances where crashes often occur and the browser is useless in retrieving/restoring.

**Comment by meerasndr** [comment]
I had to force-reboot my laptop today and was looking to restore all my tabs on Brave. Brave prompted me with a "restore" button, but this opened up a ton of really old(closed) tabs from two weeks ago. Also, going to "History" to reopen a group of tabs messes things up(sometimes opens the same bunch of tabs in different windows). I'm wondering if there is an easier way to manage crash restore.

**Comment by SeAlgoAsoma** [comment]
This is a dealkiller. I've been very happy with Brave for a few weeks but my laptop just crashed, and when I reopened Brave it offered to restore my tabs, but then did nothing when I told it to. Some of the tabs were very old, there is no way for me to manually fish them out of my history among all the other historical pages I viewed and wanted closed.

I do have "continue where you left off" selected and am upgraded to the latest version. 

This is a very nasty surprise and has destroyed weeks' worth of productivity. This bug is not something I can live with. This problem renders Brave not ready for use as a browser in real-world environments.

**Comment by NumDeP** [comment]
I second the above persons comment. Even more disappointing when you're simultaneously running normal tabs and private tabs.

Please don't go back to your muon-laptop ways. I remember the same exact thing happening and the terrible thing was similar to the above user I was using it as my primary browser to bug report issues pertaining to this and nobody did anything.

If the team is going to work holistically on **Rewards** then please tell us that this is primarily a Crypto based browser and that you'd be working on other user preferred issues along the way as opposed to focusing on them right away.

**Comment by bsclifton** [comment]
@SeAlgoAsoma definitely sorry you experienced that ☹️ Would you be able to visit brave://crashes/ and see if if the crash is recorded there? You may need to click a "send" or "submit" button, give it a few seconds, and then restart the browser. When it sends the report, it'll share a "Report ID" which you can share here which I can use to find the root cause. What OS are you on?

There are no known-to-us ways to cause this kind of loss to happen. I've tried to reproduce these kind of crashes and the data is always there. We definitely DO want to solve this problem as we strongly agree that losing anything is a bad experience ☹️ 

If anyone here experiences this, please help us with crash reports. Also, knowing the VERSION and OS that you're on is super helpful. Also helpful: if you have Rewards enabled. Windows users might be able to additionally use `eventvwr` to see if any records were written to Event Viewer around the time of the crash

**Comment by bsclifton** [comment]
@NumDeP Muon had a number of issues with session integrity which is one of the reasons we abandoned it in favor of brave-core. As shared above, I haven't been able to reproduce this. I have troubleshooted similar issues where sqlite DBs were corrupted and there was dataloss. 

In your profile directory (ex: `~/Library/Application Support/BraveSoftware/Brave-Browser/Default` on macOS) there are two files of interest:
- `Current Session`
- `Current Tabs`

It's likely one or both of these was corrupted, perhaps by crashing while a write was happening. It's definitely worth a shot trying to get the tabs back with Ctrl + Shift + T (or Cmd + Shift + T)

**Comment by krisj** [comment]
I highly recommend installing this great little extension called Session Buddy:
https://chrome.google.com/webstore/search/session%20buddy 
it saved my back with Chrome soooo many times. Should work with Brave equally well. It is the most annoying thing in the world that this is still happening!

**Comment by AndrewMorsillo** [comment]
Yep just happened to me... again. I just can't keep using Brave. No sync and it loses all my tabs is just too much to deal with in a modern browser. I really want to love Brave but with (IMO) basic/core functionality of the browser being broken it just isn't usable day-to-day no matter how much I like the other features.

**Comment by krisj** [comment]
> Yep just happened to me... again. I just can't keep using Brave. No sync and it loses all my tabs is just too much to deal with in a modern browser. I really want to love Brave but with (IMO) basic/core functionality of the browser being broken it just isn't usable day-to-day no matter how much I like the other features.

seriously - install the above plugin - works very well for me on Brave/Chrome. basically solves the problem, its a shame they can't fix the browser instead, but its been a Chrome problem for years... however with this little plugin - sorted!

**Comment by AndrewMorsillo** [comment]
I'm using Firefox for now. Container tabs are really a killer feature in FF not available elsewhere -- I love being able to log into "privacy dangerous" sites like Facebook in a tab I know is cookie/data isolated from all other browsing. Tree style tab is also a killer addon for FF not available for chrome. 

It would be awesome if BAT/Brave Rewards could be usable in other browsers via an extension rather than baked into Brave.

**Comment by steshaw** [comment]
This happened to me when my MacBook battery died. I was busy opening 3 tabs with command-click at the time. Afterwards, the "Restore" button didn't seem to do anything but disappeared when clicked. Unfortunately brave://crashes/ says "Crash reporting is disabled.". I do wonder if I may have clicked the little x button instead or bumped the Esc key which may perhaps dismiss the "Restore" button. If it's possible to accidentally dismiss the "Restore" button, is there a menu item (or such) where the session can still be restored?

**Comment by krisj** [comment]
> This happened to me when my MacBook battery died. I was busy opening 3 tabs with command-click at the time. Afterwards, the "Restore" button didn't seem to do anything but disappeared when clicked. Unfortunately brave://crashes/ says "Crash reporting is disabled.". I do wonder if I may have clicked the little x button instead or bumped the Esc key which may perhaps dismiss the "Restore" button. If it's possible to accidentally dismiss the "Restore" button, is there a menu item (or such) where the session can still be restored?

see above for the session buddy plugin - it's very nice, lightweight, actively maintained etc. But yes I agree - Brave need to sort this out. However, I am not worried about it anymore...

**Comment by meerasndr** [comment]
@steshaw I've faced incomplete restores, so I head to the `Customise and control Brave` (basically the 3 stacked horizontal lines / hamburger button) and use History -> reopen/restore x number of closed tabs. This works fine even for multiple window restores. Hope this helps.

**Comment by steshaw** [comment]
@meerasndr thanks for the tip. Unfortunately, this time "Recently Closed" is empty. I was able to trawl through my history to find a few of the pages that I could remember.

@krisj I saw your tip about session buddy when I came here and read through the comments. I ended up with a similar extension called Tab Session Manager https://github.com/sienori/Tab-Session-Manager. It's available under the MPL and also installable on Firefox and even Microsoft Edge. I'll see how that goes. The restore problem has happened a couple of times before. I headed back to Firefox the first time but Firefox is unfortunately much less snappy than Brave and Chrome.

**Comment by RgnKjnVA** [comment]
FWIW This started happening to me on startup yesterday (6/18/2020). Brave 1.10.93 on Windows 10. No tab related extensions installed. So far I can restore from History > Restore but, obviously, not ideal. Is there a bug on this yet or is this it?

**Comment by mdella** [comment]
Well, I hate to add to this thread but of course the only reason I'm here looking at the thread is that I *just* did a brave update (just prompted me yesterday so did it toay) and something like 80-100 tabs disappeared.  Just like everyone else, no way to go through my history to find them all or the session information that was used on several.  Nor any way to even remember what they all were (they were there as memory reminders of different aspects of the project I was working on).  I'm on MacOS and this last happened around 9 months ago and now again today.

Using 1.10.93 as of now. No CMD+shift+T available of course.  And the history is all over the place.  I'm just taking a break right now as my Monday morning is now going to be a couple days of restoration trying to see what I can piece together to get my work flow going again.

**Comment by scottf007** [comment]
Happened to me last night on relaunch after an upgrade. I know I should not rely on things just being there - like not saving a word document. But that is how I work, I have about 40-50 tabs....

this kills me

81a40000-1c55-8704-0000-000000000000
Windows 10 pro - latest updates I guess. 64bit

**Comment by gallegosart** [comment]
Also just updated I think yesterday (1.10.95) and had a crash that wiped out tabs and history of closed tabs. :-(
win10x64 (10.0.18363 Build 18363)
0ca90000-1c55-8704-0000-000000000000
I filed the report as #10428

**Comment by tripleee** [comment]
Not properly a crash, but a system crash followed by an upgrade. Brave was offering to upgrade, and to restore my previous session. Foolishly I chose the former, assuming I could then proceed with the latter. 40+ windows with probably 800-1000 tabs and a presentation tomorrow morning where I need to demo things I had open in the browser. /-:

**Comment by steshaw** [comment]
I switched back to [Firefox](https://www.mozilla.org/en-US/firefox/new/) a couple of weeks back as my daily driver.

**Comment by SpenMillard** [comment]
This may be useful to Mac users - I closed the browser and restored a Time Machine backup of the entire directory: Application Support > Brave Software.

When I re-opened the browser, the Restore Previous Session button appeared and did indeed restore all of my tabs from the time of yesterday's backup.

This may be a clumsy way of doing it - I read above that other users had only restored specific files, but this method did appear to work.

**Comment by nick-s-b** [comment]
I just experienced this issue with Brave for Linux. I lost all the tabs but was able to recover them because I backed up before rebooting. I think I know what the bug was in this case.

If you don't have a whole-FS backup, make sure you backup this folder: `~/.config/BraveSoftware/Brave-Browser/Default/Sessions/` It should contain 4 (or more?) files of the type `Session_<unix time>` and `Tabs_<unix time>`. Periodically copy these files somewhere or just write a script to back it up. On macOS, these files are in `~/Library/Application Support/BraveSoftware/Brave-Browser/Default/Sessions/` folder.

After the reboot and after starting Brave, I've noticed that only one tab was open and it had some image from the local disk open. I believe the issue is that somehow a second main Brave process was started when I double-clicked that file earlier in the day and this action caused a new set of Session & Tab files to be created. In my backup, I saw the old set of files (and a new one which was only few kb in size) and just replaced the new ones with those old ones from the backup. Once the Brave process is started, it will pick the newest set of Tabs and Session files and delete the older ones. This is one way of how you lose all your tabs.

Somehow, two main processes were started (even though there should be only one main process) and that caused the issue in my case. Maybe it's the root cause for many others. Brave team should re-check how they check whether another session is running and also not just delete an older set of Session & Tab files upon start... if there are multiple ones.

**Comment by janning** [comment]
I just experienced this issue with Brave for Android.
- Both the Brave-related folders ("com.brave.browser") in phone (last modified 2019-05-15) and on SD card (last modified 2020-05-24) are empty (minus the fact they each have an empty "files" sub-folder and an empty "Downloads" sub-sub-folder, but there are no files in them) ... I can look in other places for some sort of session file if I knew where else to look, but it seems rather un-intuitive to store such info elsewhere.
- Unsure how to reproduce as I was using the app just fine previously, AFAIK, and I don't recall it crashing. Just happened to open it today (2020-12-01) and suddenly none of my tabs were there. (I don't recall specifically how many, but it was only in the double-digits.) Never had this problem before.
- Recently Closed: only shows stackoverflow.com (Log In - Stack Overflow), then "Show Full History"
- History:
  -- 2020-12-01 stackoverflow.com (a specific post)
  -- 2020-12-01 stackoverflow.com (Log In - Stack Overflow)
  -- 2020-11-30 cnbc.com (FCC Chairman Ajit Pai will step...) > this was from Reddit (Open-in: Brave)
  -- the rest are 2020-11-27 and prior
- OS: Android 9; moto g(6) Build/PDS29.118-15-11-14
- Application Version: Brave 1.17.72, Chromium 87.0.4280.66

It sucks, but these things do happen. Hoping this additional info will help, even if it only means getting Brave to store session files in a safer way for a manual restore procedure.

**Comment by costaht** [comment]
Guys, I have a backup of /home/costa/.config/BraveSoftware/Brave-Browser before and after losing my tabs. Is there any information data you want me to provide that will help you to troubleshoot the problem?

I'm running:
Fedora release 33 (Thirty Three) - Linux lpt-fedora 5.9.13-200.fc33.x86_64 
Version 1.10.97 Chromium: 83.0.4103.116 (Official Build) (64-bit)

**Comment by costaht** [comment]
Just lost everything again... I'm about to go back to Firefox :'(
Here's the logs I got. I hope it helps
[brave.log](https://github.com/brave/brave-browser/files/5722818/brave.log)

**Comment by planet36** [comment]
I had to do a hard reset of my laptop.  When I opened Brave, it prompted me to restore the pages.  I clicked "Restore", but nothing happened.

I have a copy of the previous Session_##### and Tabs_##### files from the UserData/Default/Sessions/ folder.  Can those be used to restore the lost session?

**Comment by costaht** [comment]
> I had to do a hard reset of my laptop. When I opened Brave, it prompted me to restore the pages. I clicked "Restore", but nothing happened.
> 
> I have a copy of the previous Session_##### and Tabs_##### files from the UserData/Default/Sessions/ folder. Can those be used to restore the lost session?

I think you can. Close Brave, then paste your backup files overwriting the ones that are there. Open Brave again and the restore button will appear.

**Comment by planet36** [comment]
@costaht That doesn't work.

I also have a copy of the "Session Storage" folder, and within it there's a large .ldb file with a modification time before the crash.

If I kill the brave.exe process tree, is there any kind of manual intervention to make it restore what I want it to?

**Comment by costaht** [comment]
> @costaht That doesn't work.
> 
> I also have a copy of the "Session Storage" folder, and within it there's a large .ldb file with a modification time before the crash.
> 
> If I kill the brave.exe process tree, is there any kind of manual intervention to make it restore what I want it to?

That's what i do:
Close Brave, make sure isnt running in background
Replace files with my backup
Open Brave.
Brave will complain because of the new files and give you the Restore button.

Just to let you know, when I do that's i restore the backup of the entire Brave browser folder.

**Comment by planet36** [comment]
> I also have a copy of the "Session Storage" folder, and within it there's a large .ldb file with a modification time before the crash.

I checked again, and the modification time of the .ldb file is *after* the crash.  My mistake.

I've since installed Session Buddy, and it's working well so far.

**Comment by cjcole** [comment]
This has just happened for me:

Version 1.20.103 Chromium: 88.0.4324.152 (Official Build) (x86_64)

Mojave 10.14.6 (18G7016)

This was after a reboot due to laptop loss of power, not a Brave crash.

The symptoms are more or less consistent with the reports above.

Please consider bumping up the priority on this since: (1) this has been outstanding and ongoing for years now and (2) for those of us who use profiles and tabs to organize, this is a huge pain in the ass when it happens. Crashes are bad, but you can usually recover quickly and move on. Bugs like this take activity built over weeks and throw it away. It's a huge demoralizer.

Please let me know if there are any files I can gather from my installation which would help to diagnose this.

**Comment by vertigo220** [comment]
Just had this happen, too, and it's clear I'm going to have to stop using Brave. I've now lost my tabs with Brave both on mobile and desktop, which is completely unacceptable. I use multiple browsers, and Brave is the worst one when it comes to recovering from crashes. Others simply restore the tabs, whereas Brave _constantly_ gives the stupid "oops" or whatever message, requiring you to click it instead of it just doing it on its own. And when I just had to forcefully shut off my computer, when I booted back up everything was _gone_. No restore pop-up, no recently closed tabs, nothing. And at least with Firefox, on the rare occasion something like this happens, it stores backup files in a separate directory, making manual restoration at least somewhat painless.

The only thing that saved me was restoring the files from a backup, which meant I lost the last three days. At least that's not too bad, since I've only been using Brave as my secondary browser, so I've probably only lost maybe half a dozen tabs and had to close as many that I had previously closed. Still, a browser you can't depend on to keep your session safe is a useless browser.

And sure, maybe an extension "fixes" the issue, but that doesn't help all the people that don't install it because they assume the browser is going to have basic functionality that's existed in browsers for roughly 20 years, until suddenly they find that's not the case. Knowing about a solution _after the fact_ does nothing to help alleviate the frustration and massive waste of time caused by such a failure. If the Brave developers really can't/won't fix this, then they should at least display a warning on first run to use a third-party session backup extension.

**Comment by rotemg69** [comment]
Help needed. Please advise. This has just happened for me:

Current Version 1.28.105 Chromium: 92.0.4515.131 (Official Build) (arm64)

MacOS: BigSur 11.5.2

Can't recall if issue started after a Brave crash or loss of power.

The symptoms are more or less consistent with the reports above.

Please let me know if there are any files I can gather from my installation which would help to diagnose this.

**Comment by ikwyl6** [comment]
Version 1.27.111 Chromium: 2.0.4515.131 (Official Build) (64-bit)
History is blank. If I restart browser the session data with tabs get restored to something a lot older. At the same time if I try to install an extension it says 'Failed - Insufficient permissions" in the download bar at the bottom. Lots of room on drive.

**Comment by vlaidet** [comment]
@steshaw Many thanks for your solution, "Tab Session Manager" saved my day

**Comment by jnthnhng** [comment]
I was fortunate enough to not have any issue with Brave browser restoring grouped tabs until today. My computer usually sleeps, but sometime last night it shut down. When I started it up, instead of opening Brave browser which usually restores my group pages and session, I clicked on a link that opened up a web page, but none of my grouped pages loaded, nor loaded my previous session. 

I recovered my group pages by doing `CTRL+SHIFT+T`,  which reopened closed tabs. Alternatively, you can access that function by right clicking on a tab which I believe is the name thing. 

Hopefully this helps someone else.

**Comment by HuntingTarg** [comment]
### **CONTEXT:**

I came here because I was fortunate enough to find [a superuser issue](https://superuser.com/questions/1291283/how-to-restore-brave-session-manually) that shed some light on how old and chronic [this unresolved issue](https://github.com/brave/browser-laptop/issues/5512) actually is.
@silent5am @bsclifton 

 (PLEASE NOTE:  the old GitHub issue report has been archived because it was started on a build before the transition to Chromium and is no longer applicable to the current core code).

This is an issue bump and a file monkey's workaround from [reading this thread](https://github.com/brave/brave-browser/issues/4416#issuecomment-736927810) and [poking around the file structure.](https://github.com/brave/brave-browser/issues/4416#issuecomment-835936466)  This issue is sorely overdue for a fix, workaround, or just closure with reference to [third-party resources](https://github.com/brave/brave-browser/issues/4416#issuecomment-559887219) [such as mentioned above.](https://github.com/brave/brave-browser/issues/4416#issuecomment-637794064)  

### **OVERVIEW (of my issue)**

I have had crashes in the past and just decided to deal, but this time I had multiple windows open with sorted tab groups allocated to various tasks, and was going to search for similar problems and solutions one last time before transitioning away to a different primary browser (the second time in five years for me).  
What specifically occurred for me as a user was:  I had a great number of tabs open as is typical for me (low 100s).  Something went wrong and caused brave to crash, and I restarted Brave, but left the computer to do something else before returning to click the 'Restore' button on the prompt.  When I returned and clicked the button, nothing happened.  My lost session was not restored, and the dialog box went away.  This has happened to me two previous times as a Brave user, and each time it is excruciatingly frustrating.
I have a rather dated computer with only 4G of RAM, and I have experienced unrecoverable crashes of Brave associated with insufficient memory, or something I can only guess is caused by or related to a memory allocation fault.  (I trust there are enough crash reports available to the developers here to dig in further to see if my GUI dead-reckoning is accurate or not).  Only once in my experience with crashes, both restored and unrestored, have I seen a memory error warning or dialog from Windows, but it seemed an important clue, along with the fact that I have almost always had other programs running in parallel to Brave (typically Adobe Reader loading hundreds-page documents w/ graphics, and a program here on GitHub called MegaMek, which uses the Adoptium Java runtime library - not a small application for this machine).

Brave Version 1.66.110 Chromium: 125.0.6422.60 (Official Build) (64-bit)

Windows 10 Home (64 bit), version 22H2, OS build 19045.4412
@kjozwiak 
***IMPORTANT DETAIL*** My Sessions folder was copied less than an hour after the crash, and has files that predate the crash by less than one hour.

**_SO..._** here's what I tried that worked:

### **SOLUTION:** (manual workaround)

1. Close Brave
2. Copy the entire "Sessions" folder into a safe location (outside BraveSoftware/Brave-Browser/User Data); this folder is currently located under the full path 
"%drivename%/Users/%username%/AppData/Local/BraveSoftware/Brave-Browser/User Data/Default/Sessions"
for Windows 
3. _IF_ files starting with "Sessions_###" and "Tabs_###" that are timestamped before the crash can be identified,
_THEN_ compare with the working "Sessions" folder in the path identified above, and in that working folder delete all files starting with the same beginning name and timestamped after the identified files (I left "Apps_###" alone since the only occurrence was dated long before the crash and all seems to be well)
4. Relaunch Brave; the lost session or a close approximation of it should be loaded, depending on the time and amount of activity separating the last desired session save and the occurrence of the crash.

This solution or a modification of it shouldn't be hard to procure automation for; some years ago there was 'backup version' management for Windows (before the transition to SaaS - I think it was available to users for NT and XP, but not after that if memory serves), which allowed previous configurations of Windows to be loaded if something went awry after a major change.  I imagine the same functionality could be implemented here; it would likely involve changing the way that Brave handles session information, possibly to create discrete 'restore points' the way Windows did with its own configuration.  Third-party tools may not work for users on shared or unowned computers where they cannot grant such tools requisite permissions.

So I hope this can lead to a final resolution of an issue that has been lurking for (too) many years.

[Session Storage - post-crash, pre-recovery state (shortly before implementing solution).zip](https://github.com/brave/brave
[Sessions - successful recovery state.zip](https://github.com/brave/brave-browser/files/15384416/Sessions.-.successful.recovery.state.zip)

[Sessions - post-crash spare.zip](https://github.com/brave/brave-browser/files/15384413/Sessions.-.post-crash.spare.zip)

[Sessions - after-crash state.zip](https://github.com/brave/brave-browser/files/15384412/Sessions.-.after-crash.state.zip)
-browser/files/15384411/Session.Storage.-.post-crash.pre-recovery.state.shortly.before.implementing.solution.zip)

![Crash Reports (empty, in case you were wondering)](https://github.com/brave/brave-browser/assets/53635722/ee692661-967f-439e-9f65-dc82f0c9d36a)

**Comment by working-name** [comment]
July, 2024. Brave still taking a dump after you hit Restore button. 5 years to fix a bug. 🤦‍♂️

**Comment by vertigo220** [comment]
@MrKaspar This is not a support forum and is not where you should be asking this question. Also, this is Brave, not Chrome. At the very least, you should create your own issue, instead of jumping in on an unrelated one (yes, I realize it's similar, but given it's not the same exact issue, and it's not even for the correct browser, it is, in fact, unrelated). There are a lot of people subscribed to this issue, and starting, not to mention carrying on, a conversation about something else is going to trigger a lot of unwanted notifications for those people.

**Comment by MrKaspar** [comment]
Sorry.

**Comment by aimestereo** [comment]
is it just me who looses all tabs on each browser update? browser updates become more forced and annoying, but the real problem is that it now wipes open tabs

**Comment by KemonoElite** [comment]
Lost all open Tabs on a android device. Went into the history/tabs and it's fully cleaned it out. This is quite annoying to have to deal with...

**Comment by bsclifton** [comment]
Closing as stale. Nothing we can do (investigate, etc) about the previous reports without a specific way to reproduce.

If you are reading this and have ran into this recently, let's create a new issue with specific details (ex: crash details from brave://crashes tab, including crash ID). Thanks!
